Add documentation for scripts and service calls with response data (#27863)
home-assistant/home-assistant.github.io: source/_docs/scripts/service-calls.markdown

{% endraw %}

### Use templates to handle response data

> Some services may respond with data that can be used in automations. Services response data

Reviewer: ```suggestion
Some services may respond with data that can be used in automation. This data is called _service response data_. Services response data
```

> Some services may respond with data that can be used in automation. This data is called _service response data_. Services response data

Reviewer: ```suggestion
Some services may respond with data that can be used in automation. This data is called _service response data_. Service response data
```

> is typically used for data that is dynamic or large which may not be appropriate for entity state.

Reviewer: ```suggestion
is typically used for data that is dynamic or large and which may not be suited for use for entity state.
```

> is typically used for data that is dynamic or large and which may not be suited for use for entity state.

Reviewer: ```suggestion
is typically used for data that is dynamic or large and which may not be suited for use in entity state.
```

> Examples may include things like upcoming calendar events for the next week or detailed driving directions.

Reviewer: ```suggestion
Examples of service response data are upcoming calendar events for the next week or detailed driving directions.
```

Templates can also be used for handling response data. The service call can specify

> `response_variable` which is the name a [variable](https://www.home-assistant.io/docs/scripts/#variables)

Reviewer: ```suggestion
a `response_variable`. This is the [variable](https://www.home-assistant.io/docs/scripts/#variables)
```

> a `response_variable`. This is the [variable](https://www.home-assistant.io/docs/scripts/#variables)

Reviewer 2: ```suggestion
a `response_variable`. This is the [variable](/docs/scripts/#variables)
```

> that will contain the response data. This example calls a service and sets the response is stored in

Reviewer: ```suggestion
that contains the response data. You can define any name for your `response_variable`. This example calls a service and stores the response in 
```

> the variable `agenda`

Reviewer: ```suggestion
the variable called `agenda`.
```

{% raw %}

```yaml
service: calendar.list_events
target:
  entity_id: calendar.school
data:
  duration:
    hours: 24
response_variable: agenda
```

{% endraw %}

You may then use the response data in the variable `agenda` in another action
in the same script. The example below sends a notification using the response
data.

<div class='note'>

> The service call data fields depend on which notification service is used.

Reviewer: ```suggestion
Which data fields can be used in a service call depends on the type of notification service that is used.
```

</div>

{% raw %}

```yaml
service: notify.gmail_com
data:

>   target: gduser1@workspacesamples.dev
>   title: Daily agenda for {{ now().date() }}

Reviewer 2: ```suggestion
  target: "gduser1@workspacesamples.dev"
  title: "Daily agenda for {{ now().date() }}"
```
